libraries/AP_Motors/AP_MotorsMatrix.cpp

```diff
@@ -363,12 +362,249 @@ void AP_MotorsMatrix::remove_motor(int8_t motor_num)
     }
 }
 
-// remove_all_motors - removes all motor definitions
-void AP_MotorsMatrix::remove_all_motors()
+void AP_MotorsMatrix::setup_motors(motor_frame_class frame_class, motor_frame_type frame_type)
 {
-    for( int8_t i=0; i<AP_MOTORS_MAX_NUM_MOTORS; i++ ) {
+    // remove existing motors
+    for (int8_t i=0; i<AP_MOTORS_MAX_NUM_MOTORS; i++) {
         remove_motor(i);
     }
+
+    bool success = false;
+
+    switch (frame_class) {
+
+        case MOTOR_FRAME_QUAD:
+            switch (frame_type) {
+                case MOTOR_FRAME_TYPE_PLUS:
```

Comment:
I like them indented to, but it is in our rules: http://ardupilot.org/dev/docs/style-guide.html#case-statements
I'm in favor of changing it to indented, but it needs to be decided at a dev call.